File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0  # Edge case: empty list

        l = 0  # Pointer for the last unique element
        for r in range(1, len(nums)):
            if nums[r] != nums[l]:  # Found a unique element
                l += 1  # Move the unique position pointer
                nums[l] = nums[r]  # Place the unique element

        return l + 1  # Number of unique elements

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let l = 0
    for(let r = 1; r< nums.length; r++){
        if(nums[l] != nums[r]){
            l++
            nums[l] = nums[r]
          
            console.log(nums)
        }
       
    }
    return l+1
};